implementation of design and code

redmarverick opened the pull request:
- Setup webpack (Webpack Style/CSS Loader, and use HTML template)
- Create the design of the to-do list project (index.html and style.css)
- Generate a list of tasks dynamically
- config of LICENSE, README and Linters

THE FUNCTIONS OF THE WEBSITE ASIDE FROM LISTING WILL NOT WORK, AS IT IS NOT A REQUIREMENT FOR THE EXERCISE

redmarverick committed:
implementation of design and code

-  Setup webpack (Webpack Style/CSS Loader, and use HTML template)
- Create the design of the to-do list project (index.html and style.css)
- Generate a list of tasks dynamically

redmarverick committed:
Changes on readme

- changed readme from previous example

Shisui6 commented on src/index.js:
```diff
@@ -0,0 +1,39 @@
+import ListTasks from './modules/ListTasks.js';
+
+/* List Task */
+const listTask = new ListTasks();
```
According to the project requirements, by default, your app should contain a hardcoded array with some simple to-do tasks containing three keys. Then the tasks are used to populate a HTML list item. Please update your application to include this feature.

redmarverick replied:
- the listing of lasts is now hardcoded

Shisui6 reviewed (changes requested):
Hi @redmarverick, 

# Requires Changes 🍀
You have done a great job so far!👏👏
I noticed that your to-do list app is fully functional. Please note that for each project, you are supposed to implement the functionality in each milestone step by step so that you learn something new every day. Although your app is complete, it does not fulfil the requirements set for this project 

## Highlights
- Webpack setup properly ✅
- App works as expected ✅


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code._

You can also consider the following:

N/A

Cheers and Happy coding!🍻💻

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid, unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561), you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

redmarverick committed:
Changes to listing

- listing itens is now hardcoded into the code

redmarverick committed:
Changes on List and Index

AbelG101 commented on src/index.js:
```diff
@@ -0,0 +1,114 @@
+/* On submit the input */
+const task = [];
+const form = document.forms[0];
+form.addEventListener('submit', (event) => {
+  event.preventDefault();
+  if (form.newtask.value.trim().length !== 0) {
+    const item = { description: form.newtask.value, completed: false, index: task.length + 1 };
+    task.push(item);
+    display(task, item.index);
+  }
+  form.newtask.value = '';
+});
```
- [x] I like that you're thinking ahead and adding features but for this milestone, the requirement is that you just need to create a temporary array of todo task objects and render them by iterating.
- Requirement:
  ![image](https://user-images.githubusercontent.com/104968236/235934318-e3521abc-f775-4add-bb68-f2ea81ec2418.png)

- Check the example below:
```
const tasks = [
  {
    description: 'task-1',
    completed: false,
    index: 1,
  },
  {
    description: 'task-2',
    completed: false,
    index: 2,
  },
];
```
- And then you'll have to iterate and render each task.

redmarverick replied:
- manual insert functionality removed
- population of list automatically done

AbelG101 commented on src/index.js:
```diff
@@ -0,0 +1,114 @@
+  form.newtask.value = '';
+});
+
+/*
+ Save changes in Local Storage
+window.addEventListener('beforeunload', () => {
+  const savedData = { currentDescription: form.newtask.value, list: listTask.list };
+  window.localStorage.setItem('SavedData', JSON.stringify(savedData));
+});
+
+window.addEventListener('load', () => {
+  const savedData = JSON.parse(window.localStorage.getItem('SavedData'));
+
+  if (savedData.currentDescription != null) {
+    form.newtask.value = savedData.currentDescription;
+  }
+  if (savedData.list != null) {
+    for(let i = 0; i<task.length;i+=1){
+      let item = {'description':task.description, 'completed':task.completed, 'index':task.index};
+      task.push(item);
+      display(task);
+    };
+  }
+}) */
```
- [x] Please make sure to remove any commented code as this makes your codebase look unprofessional. Instead of commenting on them you can create a copy of the current file and remove the from the original.
 - This comment applies to any commented code on your project. ✔

redmarverick replied:
- comments removed

AbelG101 reviewed (changes requested):
![](https://media.giphy.com/media/7SyGkO0kQWCfnUC47Y/giphy.gif)


# Hi @redmarverick 👋, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
- Correct use of GitHub Flow ✔
-  Well-structured files ✔
- All linter checks are passing ✔
- Descriptive PR title with a great summary ✔

 ## Required Changes ♻️
- _Check the comments under the review._
- Please check the requirements carefully before you start coding. You just have to make a few changes and you're on your way 🚀.

## Optional suggestions
- N/A

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

redmarverick committed:
Changes for review

- comments removed
- manual insert functionality removed
- population of list automatically done

redmarverick committed:
Linters error correction

bobb-Rob reviewed (approved):
## STATUS: APPROVED ✔️ 🍾 🍾 🍾 🥳 🥳 🥳 

Hi @redmarverick, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

![](https://camo.githubusercontent.com/2c67944c14f5c74b9b607df18d73efb0c423dd148b3e10cb75b445a00220332e/68747470733a2f2f6d65646961322e67697068792e636f6d2f6d656469612f5061536e555a573145657970756c734f77642f67697068792e676966)

## Highlights
✅ No Linter error
✅ Correct use of GitHub flow
✅ Very descriptive Pull Request(PR)

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
